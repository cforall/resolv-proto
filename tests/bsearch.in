// Based on correspondance with Rob

bsearch E #ptr<E>
2 minus-short 2 2
3 minus-int 3 3
4 minus-long 4 4

msearch-short 2 #ptr<2>
msearch-int 3 #ptr<3>
msearch-long 4 #ptr<4>

%%

bsearch( minus(3 3) #ptr<2> )
bsearch( minus(3 3) #ptr<4> )

msearch( minus(3 3) #ptr<2> )
msearch( minus(3 3) #ptr<4> )
