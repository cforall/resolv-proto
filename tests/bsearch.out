[Void / (1,2,1,-1,0)] bsearch( minus-int( 3 3 ) => 2 #ptr<2> )
[Void / (0,2,1,-1,1)] bsearch( minus-int( 3 3 ) => 4 #ptr<4> )
[Void / (1,0,0,0,0)] msearch-short( minus-int( 3 3 ) => 2 #ptr<2> )
[Void / (0,0,0,0,1)] msearch-long( minus-int( 3 3 ) => 4 #ptr<4> )
